<launch>

<arg name="path_slam_config" default = "/home/yipuzhao/ros_workspace/package_dir/ORB_Data/"/>
<arg name="path_track_logging" default = "/mnt/DATA/tmp/ClosedNav/debug"/>
<arg name="num_all_feature" default = "1600" />
<arg name="do_rectify" default = "true" />
<arg name="do_vis" default = "0" />
<arg name="slam_pose_topic" default="/visual/pose"/>
<arg name="odom_topic" default="/odom_dummy"/>


<!-- call baselin ORB-SLAM for vision-based state estimation  -->
<node pkg="ORB_SLAM3" type="Stereo" name="visual_slam" args="$(arg path_slam_config)/ORBvoc.bin $(arg path_slam_config)/Gazebo_yaml/pinhole_stereo_orb3.yaml $(arg num_all_feature) $(arg do_rectify) $(arg do_vis) /multisense_sl/camera/left/image_raw /multisense_sl/camera/right/image_raw $(arg path_track_logging)" output="screen">
    <remap from="ORB_SLAM/camera_pose_in_imu" to="$(arg slam_pose_topic)"/>
    <remap from="/odom_dummy" to="$(arg odom_topic)"/>
</node>

</launch>
